### DISCUSSION TITLE
Filter out zero seconds visits.

### DISCUSSION THREAD

**Discussion by fabiocaccamo** [plausible/analytics#1023] [Anything Else]
## Prerequisites
- [x] I have searched open and closed issues to make sure that the bug has not yet been reported.

## Bug report
**Describe the bug**
The platform includes visits lasting zero seconds, these are probably visits from unrecognized bots or scrapers.

**Expected behavior**
Zero seconds visits should be excluded from the reports.

**Environment (If applicable):**
 - OS: any
 - Browser: any

**Comment by metmarkosaric** [comment]
What do you mean by zero seconds visits? Plausible excludes bots by default. Our visit duration only shows people who visit more than one page similar to how it's done in GA. For those who visit one page only we default to 0 seconds.

**Reply by fabiocaccamo** [reply]
<img width="1105" alt="Screenshot 2021-05-12 at 10 55 37" src="https://user-images.githubusercontent.com/1035294/117947607-a55b3a00-b310-11eb-8aaf-fb65bea09a49.png">

**Reply by fabiocaccamo** [reply]
My own website is just a single page, but visits duration is not always zero.

**Comment by metmarkosaric** [comment]
Ok. For single page apps, we support them out of the box (we also have a special script for [hash based routing](https://plausible.io/docs/hash-based-routing)). Visitors that don't interact with the page at all or don't interact with any custom events will be shown as 0 visit duration like explained earlier but those that do interact or convert on any goals will not be 0 seconds.

**Comment by fabiocaccamo** [comment]
The site I used as example is not a single page app: https://chiarazavattaro.com btw there are reported some 0 seconds visits.

**Reply by metmarkosaric** [reply]
That's normal. Our visit duration only shows people who visit more than one page. The same is case in Google Analytics. For those who visit one page only and don't click on any other pages we default to 0 seconds visit duration. You can read all the definitions here https://plausible.io/docs/metrics-definitions

**Reply by fabiocaccamo** [reply]
Ok, I understand the point now, but so why I see visits duration on my [website](https://fabiocaccamo.com/) as it is a simple single html page?
It would be nice to track the time spent on each single page.

**Reply by metmarkosaric** [reply]
Any interaction on a single page counts towards it so people that have an interaction such as clicking on something or converting on a custom goal will not be listed as 0 seconds. 

You could also test our hash based routing script on that site your hash based pages in the Top Pages report. This feature is meant for SPA but could work in your case too as you have a hash based menu https://plausible.io/docs/hash-based-routing

**Reply by fabiocaccamo** [reply]
Thank you again for the explanation and the positive feedback.

**Reply by metmarkosaric** [reply]
You're welcome! There's another [post here](https://github.com/plausible/analytics/discussions/863#discussioncomment-516551) that may give you more options for getting visit duration even from visitors that don't interact with your page.
